### PR TITLE
[988] Add support for a post selection

### PR DESCRIPTION
**Pull request by AxelRICHARD**
### Type of this PR 

- [x] New feature or improvement

### Issue(s)

https://github.com/eclipse-sirius/sirius-components/issues/988

### What does this PR do?

Introduce new concepts: WorkbenchSelection and WorkbenchSelectionEntry.
The Success of an operation can now contains a new parameter named
'newSelection' and containing a WorkbenchSelection.

### CODE REVIEW

**Review by pcdavid — COMMENTED** [review]
Any particular reason to only handle NodeTools?

**Comment by AxelRICHARD** [comment]
No reason, I will amend the commit to also handle Edges.

File: backend/sirius-components-collaborative-diagrams/src/main/resources/schema/diagram.graphqls [hunk 1]
```diff
@@ -306,7 +306,7 @@ union InvokeNodeToolOnDiagramPayload = ErrorPayload | InvokeNodeToolOnDiagramSuc
 
 type InvokeNodeToolOnDiagramSuccessPayload {
   id: ID!
-  diagram: Diagram!
+  newSelection: WorkbenchSelection!
```
Comment (sbegaudeau) [inline backend/sirius-components-collaborative-diagrams/src/main/resources/schema/diagram.graphqls:309]: The GraphQL schema of the `InvokeEdgeToolOnDiagramSuccessPayload` should be updated too (I'll do it)

File: backend/sirius-components-core/src/main/java/org/eclipse/sirius/components/core/api/WorkbenchSelection.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Used to select some objects after the execution of a tool in a representation.
+ *
+ * @author arichard
+ */
+public class WorkbenchSelection {
```
Comment (sbegaudeau) [inline backend/sirius-components-core/src/main/java/org/eclipse/sirius/components/core/api/WorkbenchSelection.java:23]: I'll add a `toString()` here

File: backend/sirius-components-core/src/main/java/org/eclipse/sirius/components/core/api/WorkbenchSelectionEntry.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+package org.eclipse.sirius.components.core.api;
+
+import java.util.Objects;
+
+/**
+ * Used to select an object after the execution of a tool in a representation.
+ *
+ * @author arichard
+ */
+public class WorkbenchSelectionEntry {
```
Comment (sbegaudeau) [inline backend/sirius-components-core/src/main/java/org/eclipse/sirius/components/core/api/WorkbenchSelectionEntry.java:22]: I'll add a `toString()` here too

File: frontend/src/diagram/DiagramWebSocketContainer.tsx [hunk 4]
```diff
@@ -867,10 +867,34 @@ export const DiagramWebSocketContainer = ({
     if (!invokeNodeToolLoading) {
       resetTools();
     }
-  }, [invokeNodeToolLoading, invokeNodeToolData, invokeNodeToolError, handleError, resetTools]);
+    if (invokeNodeToolData) {
+      const { invokeNodeToolOnDiagram } = invokeNodeToolData;
+      const newSelection = invokeNodeToolOnDiagram?.newSelection;
```
Comment (sbegaudeau) [inline frontend/src/diagram/DiagramWebSocketContainer.tsx:872]: `invokeNodeToolOnDiagram` cannot be `null` unless I am mistaken, same for `invokeEdgeToolOnDiagram`. In a similar fashion, `entries` cannot be `null` so no need for `?.` The typing should indicate that.

**Review by sbegaudeau — CHANGES_REQUESTED** [review]
I'll take over the PR to make the required changes and merge it on master

File: backend/sirius-components-collaborative-diagrams/src/main/resources/schema/diagram.graphqls (hunk 1 above)
Comment (sbegaudeau) [reply backend/sirius-components-collaborative-diagrams/src/main/resources/schema/diagram.graphqls:309]: Done

File: frontend/src/diagram/DiagramWebSocketContainer.tsx (hunk 4 above)
Comment (sbegaudeau) [reply frontend/src/diagram/DiagramWebSocketContainer.tsx]: I have added the necessary TypeScript typing for all the code manipulated in this PR.

**Review by sbegaudeau — APPROVED** [review]

**Pull request merged**